voicebox/node.py

```diff
@@ -178,52 +246,82 @@ def broadcast_audio(self, audio_stream):
                 # We were informed by the connection
                 # So we don't need to inform the connection
                 logging.info(
-                    f"Connection to {addr} Killed while Broadcasting"
+                    "Connection to %s Killed while Broadcasting",
+                    addr
                 )
                 self.end_call(addr, inform_connection=False)
                 continue
 
             connection.send_message(audio_stream, 2)
 
     def end_call(self, addr: str, inform_connection: bool = True):
+        """
+        Ends a call with a connected node.
 
+        Args:
+            addr: The address of the connected node.
+            inform_connection (bool): Whether to inform the connected node (default is True).
+        """
         connection = self.connection_pool.pop(addr, None)
 
         if not addr:
-            logging.error(f"Cannot find Address {addr} in pool")
-            return 
- 
+            logging.error("Cannot find Address %s in pool", addr)
+            return
+
         if inform_connection:
             connection.kill()
 
     @staticmethod
     def validate_connection(address: str) -> bool:
+        """
+        Validates a new connection.
+
+        Args:
+            address: The address of the potential new connection.
 
-        return True
+        Returns:
+            bool: True if the connection is valid, False otherwise.
+        """
+        return bool(address)

```

Comment:
The `validate_connection` method in the `Node` class currently returns `bool(address)`, which will always return True unless the address is None or an empty string. If there are specific conditions for a connection to be valid, they should be implemented in this method. [medium]